### PR TITLE
TriangleTypeDefinition

### PR DESCRIPTION
**Pull request by dzmitri-litvinov**
2. ТРЕУГОЛЬНИК. Заданы три числа. Выясните,  какого вида треугольник с такими длинами сторон: равносторонний, прямоугольный, равнобедренный или общего вида.

### CODE REVIEW

File: TriangleType.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+	cout << endl;
+	system("pause");
+	return 0;
+}
+
+void sideInput(int* side)
+{
+	cin >> *side;
+}
```
Comment (VadzimAlkhimionak) [inline TriangleType.cpp]: Good example of function!

File: TriangleType.cpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+void triangleInput(int *side1, int *side2, int *side3)
+{
+	cout << "Enter the 1st side length: ";
+	sideInput(side1);
+	cout << "Enter the 2nd side length: ";
+	sideInput(side2);
+	cout << "Enter the 3rd side length: ";
+	sideInput(side3);
+}
```
Comment (VadzimAlkhimionak) [inline TriangleType.cpp]: Bad example of a function. Try to think please.
I mentioned such an example in the last lesson.

File: TriangleType.cpp [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+void makeSide1Longest(int* side1, int* side2, int* side3)
+{
+	swapSidesIfRequired(side1, side2);
+	swapSidesIfRequired(side1, side3);
+}
```
Comment (VadzimAlkhimionak) [inline TriangleType.cpp:54]: We will discuss ше tomorrow

File: TriangleType.cpp [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+void swapSidesIfRequired(int *side1, int *side2)
+{
+	int temp;
+
+	if ((*side1) < (*side2))
```
Comment (VadzimAlkhimionak) [inline TriangleType.cpp]: Useless brackets ` if (*side1 < *side2)` enough

File: TriangleType.cpp [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+bool isTriangleNotExist(int side1, int side2, int side3)
+{
+	if (side1 > (side2 + side3))
+	{
+		return true;
+	}
+
+	return false;
```
Comment (VadzimAlkhimionak) [inline TriangleType.cpp]: The same `return side1 > (side2 + side3)`

File: TriangleType.cpp [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+bool isTriangleEquilateral(int side1, int side2, int side3)
+{
+	if (side1 == side2 and side2 == side3)
+	{
+		return true;
+	}
+	else { return false; }
```
Comment (VadzimAlkhimionak) [inline TriangleType.cpp]: The same `return side1 == side2 and side2 == side3`

File: TriangleType.cpp [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+bool isTriangleRight(int side1, int side2, int side3)
+{
+	if (pow(side1, 2) == (pow(side2, 2) + pow(side3, 2)))
+	{
+		return true;
+	}
+	else { return false; }
```
Comment (VadzimAlkhimionak) [inline TriangleType.cpp]: the same example above

File: TriangleType.cpp [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+bool isTriangleAcuteAngled(int side1, int side2, int side3)
+{
+	if (pow(side1, 2) < (pow(side2, 2) + pow(side3, 2)))
+	{
+		return true;
+	}
+	else { return false; }
```
Comment (VadzimAlkhimionak) [inline TriangleType.cpp]: the same example above

File: TriangleType.cpp [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+bool isTriangleIsosceles(int side1, int side2, int side3)
+{
+	if (side1 == side2 or side2 == side3)
+	{
+		return true;
+	}
+	else { return false; }
```
Comment (VadzimAlkhimionak) [inline TriangleType.cpp]: the same example above

File: TriangleType.cpp [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+void triangleTypeDefinition(int side1, int side2, int side3)
+{
+	if (isTriangleEquilateral(side1, side2, side3))
+	{
+		cout << endl << "The triangle is equilateral." << endl;
+	}
```
Comment (VadzimAlkhimionak) [inline TriangleType.cpp:98]: not suitable `if`  for this function

**Review by VadzimAlkhimionak — COMMENTED** [review]
Some comments for you :)

File: TriangleType.cpp (hunk 2 above)
Comment (dzmitri-litvinov) [reply TriangleType.cpp]: Is it possible to change in such ways?
1. Split this function into 3 functions like "side1Input", "side2Input", "side3Input". So we will have 3 functions to be used in main.
2. Just use sideInput function 3 times in main.

File: TriangleType.cpp (hunk 10 above)
Comment (dzmitri-litvinov) [reply TriangleType.cpp:98]: To tell the truth, I don't understand why it is not suitable. Where can I read about it?

File: TriangleType.cpp (hunk 2 above)
Comment (VadzimAlkhimionak) [reply TriangleType.cpp]: 2 variant

**Pull request merged**